Rakefile

```diff
@@ -151,6 +151,24 @@ namespace :build do
         HipChat.log 'Migrating <b>dashboard</b> database...'
         RakeUtils.rake 'db:migrate'
 
+        # Update the schema cache file, except for production which always uses the cache.
+        schema_cache_file = dashboard_dir('db/schema_cache.dump')
+        unless rack_env?(:production)
+          RakeUtils.rake  'db:schema:cache:dump'
+          if RakeUtils.file_changed_from_git?(schema_cache_file)
+            # Staging is responsible for committing the authoritative schema cache dump.
+            if rack_env?(:staging)
+              RakeUtils.system 'git', 'add', schema_cache_file
+              HipChat.log 'Committing updated schema_cache.dump file...', color: 'purple'
+              RakeUtils.system 'git', 'commit', '-m', '"Update schema cache dump after schema changes."'
+              RakeUtils.git_push
+            # The schema dump from the test database should always match that generated by staging.
+            elsif rack_env?(:test)
+              raise 'Unexpected database schema difference between staging and test'
```

Comment:
Do we know why this might happen? Or is this a true exception that we expect never to happen? :)